app/models/service.rb

```diff
@@ -551,8 +551,8 @@ def create_default_service_plan
   end
 
   def default_service_plan_state
-    return unless account.try(:provider_can_use?, :published_service_plan_signup)
-    account.settings.service_plans_ui_visible? ? 'hidden'.freeze : 'published'.freeze
+    return if !account || !account.provider_can_use?(:published_service_plan_signup) || !(settings = account.settings)
```

Comment:
It's getting harder to read. Better avoiding assignments inside conditionals too.

```suggestion
    return unless account && account.provider_can_use?(:published_service_plan_signup)
    settings = account.settings
    return unless settings
```